Flaky package plugin integration tests

**Bug description**
- The package plugin integration tests fails intermittently:
```
Summarizing 1 Failure:


[Fail] Package plugin integration test testing package plugin on management cluster [It] should pass all checks on management cluster 

/home/kubo/tanzu-framework/pkg/v1/tkg/test/tkgpackageclient/package_plugin_integration_test.go:579


Ran 2 of 2 Specs in 331.886 seconds

FAIL! -- 1 Passed | 1 Failed | 0 Pending | 0 Skipped

--- FAIL: TestTkgpackageclient (331.89s)

FAIL
```

**Affected product area (please put an X in all that apply)**

- ( ) APIs
- (x) Addons
- ( ) CLI
- ( ) Docs
- ( ) IAM
- ( ) Installation
- ( ) Plugin
- ( ) Security
- ( ) Test and Release
- ( ) User Experience
- ( ) Developer Experience

**Expected behavior**
- Tests should not be flaky

**Steps to reproduce the bug**

**Version** (include the SHA if the version is not obvious)

**Environment where the bug was observed (cloud, OS, etc)**

**Relevant Debug Output (Logs, manifests, etc)**

--- a/cmd/cli/plugin/package/test/package_plugin_integration_test.go
+++ b/cmd/cli/plugin/package/test/package_plugin_integration_test.go
@@ -488,21 +488,6 @@ func setUpPrivateRegistry(kubeconfigPath, clusterName string) {
 		}
 	}
 
-	By("make sure package API is available after kapp-controller restart")
-	err = retry.OnError(
-		backOff,
-		func(err error) bool {
-			return err != nil
-		},
-		func() error {
-			result = packagePlugin.ListRepository(&repoOptions)
-			if result.Error != nil {
-				return result.Error
-			}
-			return nil
-		})
-	Expect(err).ToNot(HaveOccurred())
-
 	By("make sure registry pod is running")
 	err = retry.OnError(
 		backOff,
@@ -519,6 +504,21 @@ func setUpPrivateRegistry(kubeconfigPath, clusterName string) {
 				if pod.Status.Phase != corev1.PodRunning {
 					return errors.New("registry pod is not running")
 				}
+			}
+			return nil
+		})
+	Expect(err).ToNot(HaveOccurred())
+
+	By("make sure package API is available after kapp-controller restart")
+	err = retry.OnError(
+		backOff,
+		func(err error) bool {
+			return err != nil
+		},
+		func() error {
+			result = packagePlugin.ListRepository(&repoOptions)
+			if result.Error != nil {
+				return result.Error
 			}
 			return nil
 		})

